org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/GenericEndpoint.java

```diff
@@ -42,8 +44,9 @@ protected void recursiveFindRpcMethods(Object current, Set<Class<?>> visited, Se
 			@SuppressWarnings("unchecked")
 			Function<Object, CompletableFuture<Object>> handler = (arg) -> {
 				try {
-					Object[] argument = arg == null ? new Object[0] : new Object[] { arg };
-					return (CompletableFuture<Object>) methodInfo.method.invoke(current, argument);
+					Method method = methodInfo.method;
+					Object[] arguments = this.getArguments(method, arg);
```

Comment:
Too much TS code, right? :) You could omit `this` just to align your changes with the existing code style.